src/firmware_new/boards/frankie_v1/io/proto_multicast_communication.h

```diff
@@ -0,0 +1,142 @@
+
+/**
+ * Event Flags: used to signal tasks to unblock and run specific actions
+ */
+typedef enum
+{
+    PROTO_UPDATED   = 1 << 0,  // protobuf struct has been updated
+    RECEIVED_PROTO  = 1 << 1,  // new protobuf has been received
+    RECEIVE_TIMEOUT = 1 << 2,  // blocking takes longer than NETWORK_TIMEOUT_MS
+} ProtoMulticastCommunicationEventFlags_t;
+
+/***
+ * TASKS:
+ *
+ *                    Sender Task                             Listener Task
+ *         +-------------------------------+         +------------------------------+
+ *         |                               |         |                              |
+ *         |  wait for NETIF_CONFIGURED    |         |  wait for NETIF_CONFIGURED   |
+ *         |                               |         |                              |
+ *         |  join multicast group         |         |  join multicast group        |
+ *         |                               |         |                              |
+ *     +----> wait for PROTO_UPDATED event |     +----> wait for new proto on netif |
+ *     |   |                               |     |   |                              |
+ *     |   |  acquire lock                 |     |   |  acquire lock                |
+ *     |   |                               |     |   |                              |
+ *     |   |  serialize proto to buffer    |     |   |  deserialize buffer to proto |
+ *     |   |                               |     |   |                              |
+ *     |   |  release lock                 |     |   |  release lock                |
+ *     |   |                               |     |   |                              |
+ *     +----+ send buffer to group         |     +----+ notify RECEIVED_PROTO event |
+ *         |                               |         |                              |
+ *         +-------------------------------+         +------------------------------+
+ *
+ * @param arg The pointer to the allocated communication profile to use for the Task
+ * NOTE: These functions must run as a new Task
+ */
+void io_proto_multicast_sender_Task(void* arg);
+void io_proto_multicast_listener_Task(void* arg);
+
+/**
+ * LWIP automatically places MX_LWIP_Init() in a default task. That function
+ * call internally creates another tcpip task, and the default task just loops
+ * forever.
+ *
+ * Cube forces us to generate this function, in main.c where we mark it as __weak and
+ * define the implementation in this library (proto_mutlicast_communication). When the
+ * proto communication library is linked w/ the main frankie_v1 binary, the __weak
+ * reference is resolved to the "stronger" reference to the function defined below.
+ *
+ * We take over control so we can signal other networking tasks when MX_LWIP_Init()
+ * is finished running and the network link is configured and up.
+ */
+void io_proto_multicast_startNetworkingTask(void* arg);
```

Comment:
Can we add the `void* arg` to the javadoc?

Reply:
updated to say unused